addons/web/static/src/views/list/list_renderer.js

```diff
@@ -186,6 +191,45 @@ export class ListRenderer extends Component {
             this.columns = this.getActiveColumns();
             this.withHandleColumn = this.columns.some((col) => col.widget === "handle");
         });
+        onWillStart(async () => {
+            const values = this.props.list.groups
+                ? this.props.list.groups.map((g) => g.aggregates)
+                : this.props.list.records.map((r) => r.data);
+            const currencies = new Set();
+            this.fieldCurrencies = {};
+            for (const column of this.props.archInfo.columns) {
+                const field = this.props.list.fields[column.name];
+                if (
+                    (field?.type === "monetary" && field.currency_field) ||
+                    column.widget === "monetary"
+                ) {
+                    const currencyField = this.getCurrencyField(column);
+                    this.fieldCurrencies[column.name] = new Set();
+                    for (const value of values) {
+                        if (this.props.list.groups) {
+                            value[currencyField].map((c) => {
+                                currencies.add(c);
+                                this.fieldCurrencies[column.name].add(c);
+                            });
+                        } else {
+                            currencies.add(value[currencyField]?.id);
+                            this.fieldCurrencies[column.name].add(value[currencyField]?.id);
+                        }
+                    }
+                }
+            }
+            if (currencies.size > 1) {
+                this.currencyRates = await this.orm
+                    .cache()
+                    .call("res.currency", "get_conversion_rates", [
+                        Array.from(currencies),
+                        user.activeCompany.currency_id,
+                    ]);
+            }
+        });
```

Comment:
there's a problem with doing it in willStart. It multiple currencies appear after changing the search view, you won't fetch the rates.